Userland/Libraries/LibWeb/HTML/HTMLMediaElement.cpp

```diff
@@ -1033,6 +1039,79 @@ WebIDL::ExceptionOr<void> HTMLMediaElement::pause_element()
     return {};
 }
 
+// https://html.spec.whatwg.org/multipage/media.html#dom-media-seek
+void HTMLMediaElement::seek_element(double playback_position, MediaSeekMode seek_mode)
+{
+    // FIXME: 1. Set the media element's show poster flag to false.
+
+    // 2. If the media element's readyState is HAVE_NOTHING, return.
+    if (m_ready_state == ReadyState::HaveNothing)
+        return;
+
+    // FIXME: 3. If the element's seeking IDL attribute is true, then another instance of this algorithm is already running.
+    //           Abort that other instance of the algorithm without waiting for the step that it is running to complete.
+    if (m_seeking) {
+    }
+
+    // 4. Set the seeking IDL attribute to true.
+    m_seeking = true;
```

Comment:
We can't assume we have a video element here, since this `HTMLMediaElement` (in the future) could also be a `HTMLAudioElement`. So tracking a boolean for the attribute seems fine.

Reply:
In the future, the goal is for PlaybackManager to handle both video and audio (in any combination of one or both), so it should always be there to query. If it's used for tracking multiple seek requests across threads, though, that's fine with me.